### ISSUE TITLE
Uploading files

### ISSUE THREAD

**Issue by torstenhoegel** [Slations/directus-php-sdk#8]
Hello guys,

I've been trying to get file uploads to work for a while, with no success at all. When trying to upload a file via Paw / Postman it works like a charm, but because of the `storage` and `folder` params, the file needs to be posted as the last part of the body.

I tried building some curl body's like 

```php
$body = [
    "storage" => "local",
    "folder" => "93ceada2-e52e-4073-a833-05a06979c071",
    "file" => "@" . $file,
  ];
```
with no success. Building a CURLFile object didn't help me as well.

Has anybody an working example?
Also the implementation into the php-sdk is crucial, i'd love to contribute, once smb. found a solution to this file problem.

Thanks in advance

**Comment by Slations** [comment]
Hi there,

Thanks for getting in touch, I am a one man band building an unofficial php SDK in my spare time so any help or contribution would be epic. I plan on getting this on composer shortly to make life easier for people to use but just not had the time.

I will have a go at uploading a file this afternoon after I finish work and get back to you. Will try and use the custom endpoint for sending requests through to Directus.

Kind regards,

Alan,
Slations.

**Comment by torstenhoegel** [comment]
Awesome! Thanks for looking into that :) Really this has been a struggle for me for a long time. If there's anything I can help you php-wise I'd be more than happy to help

**Comment by Slations** [comment]
@torstenhoegel no worries, would you be able to share a bit more of your code. Trying to understand where you are getting $file from?

**Comment by torstenhoegel** [comment]
Actually it comes from a posted file from a formula like $_FILES. I've tried many combinations from a curl file object to the pretty old `'@' . $file` syntax which should not be used anymore i guess..

**Comment by Slations** [comment]
Hi @torstenhoegel,

After two hours of staring at the documentation and programming I have fixed your issue, please update to v1.0.0-beta.3 where I have added two new functions and a new method to allow for calling as you have to post data as multipart not JSON. Please find example code below on how to upload a file.

```
<?php

include './Directus.php';

session_start();

// Create SDK
$directus = new Directus('https://directus.install.io');

// Auth SDK
$directus->auth_user('demo@slations.co.uk', 'Pa33w0rd');

if (isset($_POST['doUpload'])) {
	$response = $directus->files_create($_FILES["file_name"], "folder-uuid"); // If you want a folder

	$response = $directus->files_create($_FILES["file_name"]); // If you dont want a folder

	print_r($response);
} ?>
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>Test 1</title>
    </head>
    <body>
        <form method="post" enctype="multipart/form-data">
            <input type="file" name="file_name">
            <button name="doUpload">Upload</button>
        </form>
    </body>
</html>
```

Hope this sorts your issue and let me know if you need any other help :)

Kind regards,

Alan Tiller,
Slations.
